src/main/kotlin/net/maizegenetics/phgv2/pathing/MapKmers.kt

```diff
@@ -0,0 +1,97 @@
+import com.github.ajalt.clikt.parameters.groups.mutuallyExclusiveOptions
+import com.github.ajalt.clikt.parameters.groups.required
+import com.github.ajalt.clikt.parameters.groups.single
+import com.github.ajalt.clikt.parameters.options.*
+import java.io.File
+import java.util.logging.Logger
+
+/**
+ * Sealed classes to handle either a keyFile or a list of readFiles.
+
+ */
+sealed class ReadInputFile {
+    abstract fun getReadFiles() : List<KeyFileData>
+    data class KeyFile(val keyFile: String): ReadInputFile() {
+        @Override
+        override fun getReadFiles(): List<KeyFileData> {
+            check(File(keyFile).exists()) { "Key file $keyFile does not exist." }
+            val linesWithHeader = File(keyFile).bufferedReader().readLines()
+            val header = linesWithHeader.first().split("\t")
+            //convert the header into a map of column name to column index
+            val headerMap = header.mapIndexed { index, s -> s to index }.toMap()
+            check(headerMap.containsKey("sampleName")) { "Key file $keyFile must have a column named sampleName." }
+            check(headerMap.containsKey("filename")) { "Key file $keyFile must have a column named filename." }
+            return linesWithHeader.drop(1).map{lines -> lines.split("\t")}.map { linesSplit ->
+                KeyFileData(linesSplit[headerMap["sampleName"]!!], linesSplit[headerMap["filename"]!!], if(headerMap.containsKey("filename2") && linesSplit.indices.contains(headerMap["filename2"]!!)) linesSplit[headerMap["filename2"]!!] else "")
+            }
+        }
+    }
+    data class ReadFiles(val readFiles: String): ReadInputFile() {
+        @Override
+        override fun getReadFiles(): List<KeyFileData> {
+            check(readFiles.isNotEmpty()) { "--read-files must have at least one file." }
+            val fileNames = readFiles.split(",")
+            check(fileNames.size <= 2) { "--read-files must have 1 or 2 files separated by commas.  You provided: ${fileNames.size}" }
+            return listOf(KeyFileData("noSample",fileNames.first(), if(fileNames.size==1) "" else fileNames.last()))
+        }
+
+    }
+}
+
+class MapKmers : CliktCommand(help="Map Kmers to the pangenome reference") {
+
+    private val myLogger = Logger.getLogger("net.maizegenetics.phgv2.cli.MapKmers")
+
+    //./phg map-kmers \
+    //    --kmer-index kmer_index.map \
+    //    --reads my_reads.fastq \ // possibly thousands of samples being inputted
+    //    --paired
+    //    --output read_count_out.map \ // could we pipe this into impute method? // thousands of outputs
+    //    // consider batch interface here ^^
+
+    val hvcfDir by option(help = "Directory containing hvcf which build up the Haplotype Graph.")
+        .default("")
+        .validate {
+            require(it.isNotBlank()) {
+                "--hvcf-dir must not be blank"
+            }
+        }
+
+    val kmerIndex by option(help = "Kmer index file")
```

Comment:
Will you update the help to explain the format of this file?

Reply:
The format is a very complicated binary format coming from the previous step.  This will be clear in the documentation, but the user should never need to know what this file is.  I will update the help to refer back to the previous command though.